src/parameterizations/stochastic/MOM_stochastics.F90

```diff
@@ -62,69 +94,155 @@ subroutine stochastics_init(dt, grid, GV, CS, param_file, diag, Time)
+                 "dissipation estimate.", default=3, do_not_log=.not.CS%do_skeb)
+  call get_param(param_file, mdl, "SKEB_TAPER_WIDTH", taper_width, &
+                 "number of cells over which the stochastic velocity increment "//&
+                 "is tapered to zero.", default=4, do_not_log=.not.CS%do_skeb)
+  call get_param(param_file, mdl, "SKEB_USE_GM", CS%skeb_use_gm, &
+                 "If true, adds GM work rate to the SKEBS amplitude.", &
+                 default=.false., do_not_log=.not.CS%do_skeb)
+  if ((.not. CS%do_skeb) .and. (CS%skeb_use_gm)) call MOM_error(FATAL, "If SKEB_USE_GM is True "//&
+                 "then DO_SKEB must also be True.")
+  call get_param(param_file, mdl, "SKEB_GM_COEF", CS%skeb_gm_coef, &
+               "Fraction of GM work that is added to backscatter rate.", &
+               units="nondim", default=0.0, do_not_log=.not.CS%skeb_use_gm)
+  call get_param(param_file, mdl, "SKEB_USE_FRICT", CS%skeb_use_frict, &
+                 "If true, adds horizontal friction dissipation rate "//&
+                 "to the SKEBS amplitude.", default=.false., do_not_log=.not.CS%do_skeb)
+  if ((.not. CS%do_skeb) .and. (CS%skeb_use_frict)) call MOM_error(FATAL, "If SKEB_USE_FRICT is "//&
+                 "True then DO_SKEB must also be True.")
+  call get_param(param_file, mdl, "SKEB_FRICT_COEF", CS%skeb_frict_coef, &
+               "Fraction of horizontal friction work that is added to backscatter rate.", &
+               units="nondim", default=0.0, do_not_log=.not.CS%skeb_use_frict)
   call get_param(param_file, mdl, "PERT_EPBL", CS%pert_epbl, &
                  "If true, then stochastically perturb the kinetic energy "//&
                  "production and dissipation terms.  Amplitude and correlations are "//&
                  "controlled by the nam_stoch namelist in the UFS model only.", &
                  default=.false.)
-  if (CS%do_sppt .OR. CS%pert_epbl) then
-    num_procs = num_PEs()
-    allocate(pelist(num_procs))
-    call Get_PElist(pelist,commID = mom_comm)
-    pe_zero = root_PE()
-    nx = grid%ied - grid%isd + 1
-    ny = grid%jed - grid%jsd + 1
-    call init_stochastic_physics_ocn(dt,grid%geoLonT,grid%geoLatT,nx,ny,GV%ke, &
-                                     CS%pert_epbl,CS%do_sppt,pe_zero,mom_comm,iret)
-    if (iret/=0)  then
-      call MOM_error(FATAL, "call to init_stochastic_physics_ocn failed")
-    endif
-
-    if (CS%do_sppt) allocate(CS%sppt_wts(grid%isd:grid%ied,grid%jsd:grid%jed), source=0.0)
-    if (CS%pert_epbl) then
-      allocate(CS%epbl1_wts(grid%isd:grid%ied,grid%jsd:grid%jed), source=0.0)
-      allocate(CS%epbl2_wts(grid%isd:grid%ied,grid%jsd:grid%jed), source=0.0)
-    endif
-  endif
-  if (CS%do_sppt) then
-    CS%id_sppt_wts = register_diag_field('ocean_model', 'sppt_pattern', CS%diag%axesT1, Time, &
-         'random pattern for sppt', 'None')
+
+  if (CS%do_sppt .OR. CS%pert_epbl .OR. CS%do_skeb) then
+     num_procs = num_PEs()
```

Comment:
Please follow the 2-space MOM6 indenting convention with this revised code, as described at https://github.com/mom-ocean/MOM6/wiki/Code-style-guide#white-space .